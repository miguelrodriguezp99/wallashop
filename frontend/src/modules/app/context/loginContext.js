import { createContext, useState } from "react";

// Este es el que tenemos que consumir
export const LogginContext = createContext();

// Este es el que nos provee de acceso al contexto
export function LogginProvider({ children }) {
	const [token, setToken] = useState(null);
	const [image, setImage] = useState(null);
	const [user, setUser] = useState({});

	return (
		<LogginContext.Provider
			value={{
				token,
				setToken,
				image,
				setImage,
				user,
				setUser,
			}}
		>
			{children}
		</LogginContext.Provider>
	);
}
