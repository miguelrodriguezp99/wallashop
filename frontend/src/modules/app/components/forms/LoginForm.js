import { Button } from "@nextui-org/react";
import React, { useContext, useState } from "react";
import { useNavigate } from "react-router-dom";
import { toast } from "sonner";
import { login } from "../../../../backend/userService.js";
import { config } from "../../../../config/constants.js";
import { LogginContext } from "../../context/loginContext.js";
import Input from "../inputs/LoginInput.js";

const LoginForm = () => {
	const [username, setUsername] = useState("");
	const [password, setPassword] = useState("");
	const { setToken, setImage, setUser } = useContext(LogginContext);

	const navigate = useNavigate();

	//CAMBIAR TODA ESTA LOGICA A OTRO SITIO -----------------------------
	const onSuccess = (authenticatedUser) => {
		setImage(authenticatedUser.user.avatar);
		setToken(authenticatedUser.serviceToken);
		setUser(authenticatedUser.user);

		toast.success("Logeado correctamente.");
		localStorage.setItem(
			config.SERVICE_TOKEN_NAME,
			`Bearer ${authenticatedUser.serviceToken}`,
		);
		localStorage.setItem("user", JSON.stringify(authenticatedUser.user));
		localStorage.setItem("avatar", authenticatedUser.user.avatar);
		navigate("/wallashop/");
	};

	const onErrors = (error) => {
		toast.error("Error al logearse.");
	};

	const reauthenticateUser = () => {
		console.log(
			"Se requiere reautenticación. Mostrar cuadro de diálogo de inicio de sesión.",
		);
	};
	// ------------------------------------------------------------------

	const handleSubmit = () => {
		login(username, password, onSuccess, onErrors, reauthenticateUser);
	};

	return (
		<>
			<div className="flex items-center justify-center overflow-hidden h-[calc(100vh-65px)]">
				<div className="border border-solid border-gray-200 rounded-2xl text-center text-gray-200 font-helvetica py-10 px-16 w-300 mb-80 bg-black mt-60 overflow-hidden hover:border-blue-300">
					<h1 className="text-7xl font-bold bg-gradient-to-r from-blue-500 to-blue-700 text-transparent bg-clip-text">
						LOG IN
					</h1>
					<form onSubmit={() => handleSubmit()}>
						<label>
							<Input
								type="text"
								name="username"
								value={username}
								placeholder={"Username"}
								fn={setUsername}
							/>
						</label>
						<label>
							<Input
								label="Password"
								name="password"
								type={"password"}
								value={password}
								placeholder={"Password"}
								fn={setPassword}
							/>
						</label>
						<div className="form-buttons">
							<Button
								className="text-white ml-2 mr-2"
								onPress={() => handleSubmit()}
								as={Button}
								color="primary"
								href="#"
								variant="ghost"
							>
								Log in
							</Button>

							<Button
								className="text-white ml-2 mr-2"
								onPress={() => navigate("/wallashop/home")}
								as={Button}
								color="primary"
								href="#"
								variant="ghost"
							>
								Home
							</Button>
						</div>
					</form>
				</div>
			</div>
		</>
	);
};

export default LoginForm;
