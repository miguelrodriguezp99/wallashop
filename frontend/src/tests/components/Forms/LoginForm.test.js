import { render } from '@testing-library/react';
import LoginForm from '../../../modules/app/components/forms/LoginForm';
import { LogginContext } from '../../../modules/app/context/loginContext'; 
import { MemoryRouter } from 'react-router-dom';

describe('LoginForm component', () => {
  it('matches snapshot', () => {
    const fakeLogginContext = {
      login: jest.fn(),
      setToken: jest.fn(),
      setUser: jest.fn(),
      setImage: jest.fn(),
    };

    const { asFragment } = render(
      <MemoryRouter>
        <LogginContext.Provider value={fakeLogginContext}>
          <LoginForm />
        </LogginContext.Provider>
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
  
});