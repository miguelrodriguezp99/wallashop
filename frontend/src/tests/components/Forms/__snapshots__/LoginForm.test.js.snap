// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex items-center justify-center overflow-hidden h-[calc(100vh-65px)]"
  >
    <div
      class="border border-solid border-gray-200 rounded-2xl text-center text-gray-200 font-helvetica py-10 px-16 w-300 mb-80 bg-black mt-60 overflow-hidden hover:border-blue-300"
    >
      <h1
        class="text-7xl font-bold bg-gradient-to-r from-blue-500 to-blue-700 text-transparent bg-clip-text"
      >
        LOG IN
      </h1>
      <form>
        <label>
          <div
            class="form__group field"
          >
            <input
              class="form__field"
              name="username"
              placeholder="Username"
              type="text"
              value=""
            />
            <label
              class="form__label"
              for="name"
            >
              Username
            </label>
          </div>
        </label>
        <label>
          <div
            class="form__group field"
          >
            <input
              class="form__field"
              name="password"
              placeholder="Password"
              type="password"
              value=""
            />
            <label
              class="form__label"
              for="name"
            >
              Password
            </label>
          </div>
        </label>
        <div
          class="form-buttons"
        >
          <button
            class="group tap-highlight-transparent transition-transform-colors-opacity data-[hover=true]:opacity-hover z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 border-medium bg-transparent px-unit-4 min-w-unit-20 h-unit-10 text-small gap-unit-2 rounded-medium [&>svg]:max-w-[theme(spacing.unit-8)] data-[pressed=true]:scale-[0.97] transition-transform-colors-opacity motion-reduce:transition-none border-primary hover:!text-primary-foreground hover:!bg-primary text-white ml-2 mr-2"
            href="#"
            role="button"
            tabindex="0"
            type="button"
          >
            Log in
          </button>
          <button
            class="group tap-highlight-transparent transition-transform-colors-opacity data-[hover=true]:opacity-hover z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 border-medium bg-transparent px-unit-4 min-w-unit-20 h-unit-10 text-small gap-unit-2 rounded-medium [&>svg]:max-w-[theme(spacing.unit-8)] data-[pressed=true]:scale-[0.97] transition-transform-colors-opacity motion-reduce:transition-none border-primary hover:!text-primary-foreground hover:!bg-primary text-white ml-2 mr-2"
            href="#"
            role="button"
            tabindex="0"
            type="button"
          >
            Home
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
